luigi/parameter.py

```diff
@@ -244,16 +244,17 @@ def _parse_list(self, xs):
         else:
             return self._batch_method(map(self.parse, xs))
 
-    def serialize(self, x):
+    def serialize(self, x, name=None):
         """
         Opposite of :py:meth:`parse`.
 
         Converts the value ``x`` to a string.
 
         :param x: the value to serialize.
+        :param name: the parameter name, for reporting purposes in case of an error.
         """
         if not isinstance(x, six.string_types) and self.__class__ == Parameter:
-            warnings.warn("Parameter {0} is not of type string.".format(str(x)))
+            warnings.warn("Parameter {0} is not of type string.".format(name or x))
```

Comment:
In this function yes. And also in the overriden version in TimeDeltaParameter.serialize